protoc-gen-go/generator/generator.go

```diff
@@ -1573,7 +1573,9 @@ func (g *Generator) goTag(message *Descriptor, field *descriptor.FieldDescriptor
 		if *field.Type == descriptor.FieldDescriptorProto_TYPE_BYTES {
 			name += ",proto3"
 		}
-
+		if *field.Type == descriptor.FieldDescriptorProto_TYPE_STRING {
+			name += ",utf8"
```

Comment:
Should this use the "proto3" tag instead? We may discover other cases in the future where proto2/proto3 fields need different handling, and it'll be more efficient to have a single tag in that case.

If so, should we preemptively tag all proto3 fields, even if they currently have no semantic difference?

Reply:
The `proto3` tag was already added to distinguish the semantic difference for `[]byte{}` vs `[]byte(nil)` when it comes to the bytes type.

Looking through the code history, the attribute was only generated on the bytes type out of fear for binary size increase. If we want to go with `proto3`, should we just generate it for all fields? That's an increase of 7 bytes for every field. Alternatively, we can only generate it for the `string` field.

Reply:
If we're worried about binary size increase that argues for a proto3 tag rather than utf8, to avoid the possibility of redundant tags in the future.

I don't really have a strong opinion on every field vs. string, although there are a couple arguments for the former: It avoids any possibility of needing to do this again in the future, and I think all the cases where 7 bytes per field might conceivably be an issue are proto2 anyway.

Reply:
I enabled it unilaterally for all fields.